File: ChatClientVB/Services/DialogService.vb
Imports Microsoft.Win32

Public Class DialogService
    Implements IDialogService

    Public Sub ShowNotification(message As String, Optional caption As String = "") Implements IDialogService.ShowNotification
        MessageBox.Show(message, caption)
    End Sub

    Public Function OpenFile(caption As String, Optional filter As String = "All files (*.*)|*.*") As String Implements IDialogService.OpenFile
        Dim diag As New OpenFileDialog
        diag.InitialDirectory = Environment.GetFolderPath(Environment.SpecialFolder.MyPictures)
        diag.Title = caption
        diag.Filter = filter
        diag.CheckFileExists = True
        diag.CheckPathExists = True
        diag.RestoreDirectory = True

        If diag.ShowDialog() = True Then Return diag.FileName
        Return String.Empty
    End Function

    Public Function ShowConfirmationRequest(message As String, Optional caption As String = "") As Boolean Implements IDialogService.ShowConfirmationRequest
        Dim result = MessageBox.Show(message, caption, MessageBoxButton.OKCancel)
        Return result.HasFlag(MessageBoxResult.OK)
    End Function
End Class

File: ChatClientCS/Services/DialogService.cs
using System;
using Microsoft.Win32;
using System.Windows;

namespace ChatClientCS.Services
{
    public class DialogService : IDialogService
    {
        public string OpenFile(string caption, string filter = "All files (*.*)|*.*")
        {
            OpenFileDialog diag = new OpenFileDialog();
            diag.InitialDirectory = Environment.GetFolderPath(Environment.SpecialFolder.MyPictures);
            diag.Title = caption;
            diag.Filter = filter;
            diag.CheckFileExists = true;
            diag.CheckPathExists = true;
            diag.RestoreDirectory = true;

            if (diag.ShowDialog() == true) return diag.FileName;
            return string.Empty;
        }

        public bool ShowConfirmationRequest(string message, string caption = "")
        {
            var result = MessageBox.Show(message, caption, MessageBoxButton.OKCancel);
            return result.HasFlag(MessageBoxResult.OK);
        }

        public void ShowNotification(string message, string caption = "")
        {
            MessageBox.Show(message, caption); ;
        }
    }
}
